sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift

```diff
@@ -0,0 +1,348 @@
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCommunication
+import AzureCore
+import Foundation
+
+public class ChatThreadClient {
+    // MARK: Properties
+
+    public let threadId: String
+    private let endpoint: String
+    private let credential: CommunicationUserCredential
+    private let options: AzureCommunicationChatClientOptions
+    private let service: AzureCommunicationChatService
+
+    // MARK: Initializers
+
+    /// Create a ChatThreadClient.
+    /// - Parameters:
+    ///   - endpoint: The Communication Services endpoint.
+    ///   - credential: The user credential.
+    ///   - threadId: The chat thread id.
+    ///   - options: Options used to configure the client.
+    public init(
+        endpoint: String,
+        credential: CommunicationUserCredential,
+        threadId: String,
+        withOptions options: AzureCommunicationChatClientOptions
+    ) throws {
+        self.threadId = threadId
+        self.endpoint = endpoint
+        self.credential = credential
+        self.options = options
+
+        guard let endpointUrl = URL(string: endpoint) else {
+            throw AzureError.client("Unable to form base URL")
+        }
+
+        let authPolicy = CommunicationUserCredentialPolicy(credential: credential)
+
+        let client = try AzureCommunicationChatClient(
+            endpoint: endpointUrl,
+            authPolicy: authPolicy,
+            withOptions: options
+        )
+
+        self.service = client.azureCommunicationChatService
+    }
+
+    // MARK: Public Methods
+
+    /// Updates the ChatThread's properties.
+    /// - Parameters:
+    ///   - chatThread: Request for updating a chat thread.
+    ///   - options: Update chat thread options.
+    ///   - completionHandler: A completion handler that receives a status code on success.
+    public func update(
+        chatThread: UpdateChatThreadRequest,
```

Comment:
Consider accepting a `ChatThread` object unless the `UpdateChatThreadRequest` object has additional properties.

Reply:
Thanks, I checked and this is actually supposed to be renamed to updateTopic() or similar. 

Regards to making everything consistent,

As an example for send methods we have:
`public func send(readReceipt: SendReadReceiptRequest, withOptions...)`
`public func send(message: SendChatMessageRequest, withOptions...)`
and 
`public func sendTypingNotification(withOptions...)` where there's no "typing notification" argument, just options, similar to the list methods - listMessages, listParticipants etc.

If we go with the format of `getChatThread(withId...)` do we still leave `send(message...)` or change to `sendMessage(message...)` ie should I make them ALL consistent or only where it isn't redundant?

Reply:
I've update `getChatThread(withId) -> get(chatThreadId)` and updated the getMessage to be consistent as well for now